### DISCUSSION TITLE
Getting event id from microsoft logging extension integration

### DISCUSSION THREAD

**Discussion by HaydnDias** [getsentry/sentry-dotnet#1331] [Q&A] [answered]
Hi, we're using sentry with microsoft logging by catching errors and calling ILogger.LogError(...etc), but we'd like to be able to get the event id for that log to return with api's/provide to consumers for tracing and reporting, is there a simple way to get this in the same context reliably?

**Comment by bruno-garcia** [comment, marked as answer]
You can get the id via `SentrySdk.LastEventId`. This value is backed by a `AsyncLocal` so it should include the value of the event id you have just captured, e.g:

```
_logger.LogError(..);
var sentryId = SentrySdk.LastEventId;
```

if you don't want to couple your code with the static class `SentrySdk` you can take via DI `IHub`. [More on the docs](https://docs.sentry.io/platforms/dotnet/unit-testing/). 

https://github.com/getsentry/sentry-dotnet/blob/ad9f5ff4ebeb1620d42d781b15d0e9f7edec441d/src/Sentry/IHub.cs#L23

But since you're already taking `IHub` via DI (or using SentrySdk), an alternative is:

```
var sentryId = _hub.CaptureException(ex);
```

**Reply by HaydnDias** [reply]
That's great, thank you!

**Reply by HaydnDias** [reply]
So I gave this a try, after an exception was logged with `_logger.LogError(...);`, but the result came up as `00000000000000000000000000000000`, I can see that the error was logged in sentry. Any ideas?
![image](https://user-images.githubusercontent.com/7255658/144030955-defba3c5-1652-4b9b-b89e-6d3efaf89a95.png)

**Reply by bruno-garcia** [reply]
Is the event captured?

Calling `LogError` will call Sentry's `CaptureEvent`:

https://github.com/getsentry/sentry-dotnet/blob/41cf450f3000f7a2a0311d85011a4ca5bbacd56a/src/Sentry.Extensions.Logging/SentryLogger.cs#L102

Is it possible the SDK wasn't enabled? Or the Hub you got in there isn't the one bound to the `SentrySdk` static class?

As a test, can you check if `SentrySdk.LastEventId` has the value? And confirm that the event is getting to Sentry?

Also for testing, just try: `_hub.CaptureException(..)` instead of `LogError`.
